docs/src/main/docs/health/02_health_in_k8s.adoc

```diff
@@ -0,0 +1,217 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Kubernetes probes
+:description: Kubernetes probes
+:keywords: helidon, readiness, liveliness, probes, kubernetes
+:kubernetes-probes-url: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes
+
+This document describes usages of the Helidon health check API with Kubernetes.
+
+== About Kubernetes probes
+
+Probes is the term used by Kubernetes to describe health checks for containers
+ (more info link:{kubernetes-probes-url}[here]).
+
+There are two types of probes:
+
+* `liveness`: Indicates whether the container is running
+* `readiness`: Indicates whether the container is ready to service requests
+
+Probes can be implemented with three different mechanisms:
+
+. running a command inside a container
+. making an `HTTP` request against a container
+. opening a `TCP` socket against a container
+
+A microservice exposed to HTTP will typically implement both the liveness probe
+ and the readiness probe using HTTP requests.
+
+Probes can be configured with different parameters, see the most relevant
+ ones below:
+
+[cols="2,5",role="flex, sm7"]
+|=======
+| `initialDelaySeconds`
+| Number of seconds before probes are initiated
+
+| `periodSeconds`
+| How often to perform the probe. Default to 10 seconds. Minimum value is 1
+
+| `timeoutSeconds`
+| Number of seconds after which the probe times out. Defaults to 1 second.
+ Minimum value is 1
+
+| `failureThreshold`
+| Number of failed probes before giving up. Defaults to 3. Minimum value is 1.
+|=======
+
+=== Liveness probe
+
+The liveness probe is used to verify the container has become unresponsive,
+ for example it can detect deadlocks or analyze heap usage. When Kubernetes
+ gives up on a liveness probe, the corresponding pod is restarted.
+
+NOTE: A bad liveness probe can result in repeated restarts.
```

Comment:
What does "bad" mean here? Too low failure threshold or something else?

Reply:
Bad probe means poorly implemented:
 - e.g. checks all dependencies strictly and does fail with temporary transient issue
Or also or poorly configured (timeouts and interval not big enough)

Reply:
I suggest:

"NOTE: The liveness probe can result in repeated restarts in certain cases. For example, if the probe is implemented to check all the dependencies strictly, then it can fail repeatedly for temporary issues. Repeated restarts can also occur if `timeoutSeconds` or `periodSeconds` is too low."

Reply:
I've made the suggested change.